content/blog/2015-06-20-monitoring-dreamhack.md

```diff
@@ -0,0 +1,122 @@
+
+## The Architecture
+The monitoring solution consists of three layers: collection, storage, presentation. Our most critical collectors are snmpcollector (SNMP) and ipplan-pinger (ICMP), closely followed by dhcpinfo (DHCP lease stats). We also have some scripts that dump stats about other systems into node_exporter's textfile collector.
+
+[![dhmon Architecture](http://i.imgur.com/6gN3MRp.png)](http://i.imgur.com/6gN3MRp.png)
+<center>*The current architecture plan of dhmon as of Summer 2015*</center>
+
+As for storage we use Prometheus as a central timeseries storage and querying engine but we also use Redis and memcache to export snapshot views of binary information that we collect but cannot store in Prometheus in any sensible way or when we need to access very fresh data.
+
+One such case is in our presentation layer. We use our dhmap web application to get an overview of the overall health of the access switches. In order to be effective at resolving errors we need a latency of ~10 seconds from data collection to presentation. Our goal is to have fixed the problem before the customer notices, or at least before he/she has walked over to the support people to report an issue. For this reason, we have been using memcache since the beginning to access the latest snapshot of the network.
+
+We continued to use memcache this event as well for our low-latency data while using Prometheus for everything that's historical or not as latency sensitive. This decision was made simply because we were unsure how Prometheus would perform. In the end we found no reason for why we can't use Prometheus for this data as well - we will definitely try to replace our memcache with Prometheus next Dreamhack.
+
+[![dhmon Visualization](http://i.imgur.com/D5I0Ztbl.png)](http://i.imgur.com/D5I0Ztb.png)
+<center>*The overview of our access layer visualized by dhmon*</center>
+
+## Prometheus Setup
+The block that so far has been referred to as Prometheus really consists of three products: Prometheus, PromDash and Alertmanager. The setup is fairly basic and all three softwares are running on the same host. Everything is served by an Apache web server that just acts as a reverse proxy.
+
+    ProxyPass /prometheus http://localhost:9090/prometheus
+    ProxyPass /alertmanager http://localhost:9093/alertmanager
+    ProxyPass /dash http://localhost:3000/dash
+
+## Exploring the Network
+Prometheus is a competent querying engine, and powered by streaming information from all over your network allows you to do pretty cool things. However sometimes the queries need to process too much data to finish within a reasonable amount of time. This happened to us when we wanted to graph the top 5 utilized links. While the query worked it would take about the time we set our timeout limit to, meaning it was both slow and flaky. We decided to try to use Prometheus' precomputing facility.
```

Comment:
I had a hard time reading the first sentence. Maybe:

Prometheus *has* a competent querying engine *that*, powered by streaming information from all over your network, allows you to do pretty cool things.